go/api/transaction.go

```diff
@@ -0,0 +1,240 @@
+// Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0
+
+package api
+
+// #include "../lib.h"
+// void successCallback(void *channelPtr, struct CommandResponse *message);
+// void failureCallback(void *channelPtr, char *errMessage, RequestErrorType errType);
+import "C"
+
+import (
+	"fmt"
+	"unsafe"
+
+	"github.com/valkey-io/valkey-glide/go/api/config"
+	"github.com/valkey-io/valkey-glide/go/api/errors"
+	"google.golang.org/protobuf/proto"
+)
+
+type Transaction struct {
+	*baseClient // Embed baseClient to inherit all methods like Get
+	*GlideClient
+	*GlideClusterClient
+	commands []Cmder
+}
+
+type Cmder interface {
+	Name() C.RequestType
+	Args() []string
+	Route() config.Route
+}
+
+type GenericCommand struct {
+	name  C.RequestType
+	args  []string
+	route config.Route
+}
+
+func (cmd *GenericCommand) Name() C.RequestType {
+	return cmd.name
+}
+
+func (cmd *GenericCommand) Args() []string {
+	return cmd.args
+}
+
+func (cmd *GenericCommand) Route() config.Route {
+	return cmd.route
+}
+
+func NewMultiCommand() Cmder {
+	return &GenericCommand{name: C.Multi, args: []string{}}
+}
+
+func NewExecCommand() Cmder {
+	return &GenericCommand{name: C.Exec, args: []string{}}
+}
+
+// Override sendCommand to queue commands in the transaction
+func (t *Transaction) sendCommand(requestType C.RequestType, args []string) (*C.struct_CommandResponse, error) {
+	t.commands = append(t.commands, &GenericCommand{name: requestType, args: args})
+	return nil, nil // Queue the command instead of executing immediately
+}
+
+// Override sendCommandWithRoute to queue commands in the transaction
+func (t *Transaction) sendCommandWithRoute(
+	requestType C.RequestType,
+	args []string,
+	route config.Route,
+) (*C.struct_CommandResponse, error) {
+	t.commands = append(t.commands, &GenericCommand{name: requestType, args: args, route: route})
+	return nil, nil
+}
+
+// Exec executes all queued commands as a transaction
+func (t *Transaction) Exec() ([]any, error) {
```

Comment:
This method should have a `context.Context` parameter to support cooperative cancellation, tracing hooks, etc.
 
 In fact, every method that causes networking resources to be consumed should accept a Context, which would include every method of every *Commands interface.

Reply:
@marcind We are doing some internal discussion to see what is possible, and what the timeframes look like. I will post more details on the issue as we get agreement.  At this point it's a question of how complete of an implementation we are able to do, and what kind of schedule impact is involved. I am hopeful that we can do a reasonable implementation for a 2.0 launch.